src/platform_impl/macos/window_delegate.rs

```diff
@@ -821,7 +824,15 @@ impl WindowDelegate {
     }
 
     pub fn set_transparent(&self, transparent: bool) {
-        self.window().setOpaque(!transparent)
+        self.window().setOpaque(!transparent);
+
+        let color = if transparent {
+            unsafe { NSColor::clearColor() }
+        } else {
+            unsafe { NSColor::windowBackgroundColor() }
+        };
```

Comment:
Better document why `setOpaque` is not enough.

```suggestion
        // AppKit draws the window with a background color by default, which is usually really
        // nice, but gets in the way when we want to allow the contents of the window to be
        // transparent, as in that case, the transparent contents will just be drawn on top of
        // the background color. As such, to allow the window to be transparent, we must also set
        // the background color to one with an empty alpha channel.
        let color = if transparent {
            // Let's use `clearColor`, the alpha channel is 0.0 on that.
            unsafe { NSColor::clearColor() }
        } else {
            // Otherwise, reset to the default `windowBackgroundColor`.
            unsafe { NSColor::windowBackgroundColor() }
        };
```